server/lib/realtime/workflows/execution_manager.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Realtime.Workflows.ExecutionManager do
+
+  require Logger
+
+  alias Realtime.Workflows.TransientExecutionManager
+  alias Realtime.Workflows.PersistentExecutionManager
+
+  def start_workflow_execution(workflow, execution, opts \\ []) do
```

Comment:
@fracek

> Workflows can be transient or persistent. Transient workflows are executed inside a Task, if the realtime server restarts or if they crash they are not restarted. Persistent workflows are stored as Oban jobs, they are persisted across restarts.

I was just wondering why transient workflows aren't inserted to the Oban queue as well. 

Here's how I thought about it: both transient and persistent workflows are inserted into Oban queue, and `is_persistent` is passed to `args` when inserting to Oban as well. When `Oban.Worker` processes a job, it'll route the args to a transient `Task` or a persistent `:gen_statem` (assuming you're going to use it for the interpreter) based on `args.is_persistent`.

I bring this up b/c if a user kicks off a one off workflow, and `realtime` crashes and restarts, then the user won't really know what happened to the executed workflow. If we alert them that something went wrong then they'll have to manually kick off the same one off workflow? Seems like something we can take care of server-side and make sure it's successfully completed.

However, I might be misunderstanding something, especially since I don't know how you're going to build the interpreter yet. Just thought I throw this out there!

Reply:
The idea behind transient workflows is that we don't touch the database, the trade-off is that if the server crashes or is updated the execution is lost. The uses cases we discussed with @kiwicopple were as follows:

* Transient: for example IoT applications or starting the workflow from the rest API. In the first case we don't care about the individual execution since we're probably going to get hundreds more in the next few seconds, in the second case we don't care because the http handler was killed together with the server so we won't be able to send the response back.
* Persistent: anything else. The idea is to use oban to help us build an interpreter that can span multiple restarts. I'm thinking for example to use the scheduling functionality to implement the `Wait` state.

More precisely, we touch the database only once (or twice if we include information about the workflow termination) with transient workflows, while for persistent workflows we touch it once (or twice) + 3 times for each state (1 to insert the job, 1 to change state to executing, 1 to change state to completed).

Hope what I wrote makes sense, if not let me know and I can expand more.